validation/citrus-validation-json/src/main/java/com/consol/citrus/validation/json/JsonTextMessageValidator.java

```diff
@@ -71,15 +71,19 @@
     @SuppressWarnings("unchecked")
     public void validateMessage(Message receivedMessage, Message controlMessage,
                                 TestContext context, JsonMessageValidationContext validationContext) {
-        if (controlMessage == null || controlMessage.getPayload() == null) {
-            log.debug("Skip message payload validation as no control message was defined");
-            return;
-        }
 
         log.debug("Start JSON message validation ...");
 
         if (validationContext.isSchemaValidationEnabled()) {
-            performSchemaValidation(receivedMessage, context, validationContext);
+            jsonSchemaValidation.validate(receivedMessage, context, validationContext);
+        }
+
+        // TODO: @Christoph do you have any doubts about moving this check behind the schema validation?
```

Comment:
Can we revert this change? Now with the new solution where the schema validator performs the validation on send operation instead of the message validator?

Reply:
for backward compatibility reasons. we can do this with the next major release though